src/cxy_ctl.erl

```diff
@@ -653,3 +666,27 @@ get_buffer_times(Buffer_Name, Num_Items) ->
 format_buffer_times({Task_Fun, {_,_,Micro} = Start, Spawn_Elapsed, Exec_Elapsed}) ->
     [{task_fun, Task_Fun}, {start, calendar:now_to_universal_time(Start), Micro},
      {spawn_time_micros, Spawn_Elapsed}, {exec_time_micros, Exec_Elapsed}].
+
+-define(HW_READ_CMD, {?HIGH_WATER_PROCS_POS, 0}).
+-define(HW_RESET_CMD, {?HIGH_WATER_PROCS_POS, -1, 0, 0}).
+
+%% @doc
+%%   Return the highest number of concurrent processes for a given task type.
+%%   The one-argument form just returns the high-water value, while the
+%%   two-argument form allows the value to be reset.
+%% @end
+
+-spec high_water(task_type()) -> non_neg_integer().
+-spec high_water(task_type(), cxy_clear()) -> non_neg_integer().
+
+high_water(Task_Type) ->
+  high_water(Task_Type, no_clear).
+
+high_water(Task_Type, ClearCmd) ->
+    case ClearCmd of
+        clear ->
+            [Old_High_Water, _] = ets:update_counter(?MODULE, Task_Type, [?HW_READ_CMD, ?HW_RESET_CMD]),
```

Comment:
After reset the second counter value must always be 0, right? No reason not to fail immediately when something is unexpected.